Userland/Libraries/LibGUI/Notification.cpp

```diff
@@ -60,15 +65,26 @@ Notification::~Notification()
 void Notification::show()
 {
     VERIFY(!m_showing);
+    VERIFY(!m_disposed);
+    if (!m_connection->is_connected()) {
+        // This would imply that the NotificationServer crashed before we could send it any data.
+        VERIFY_NOT_REACHED();
+    }
```

Comment:
This edge case can be avoided by merging `m_showing`, `m_disposed`, and `m_connection` into `RefPtr<> m_connection` that contains a connection if and only if it's currently showing. In other words: The connection is only created on-demand.